docs/proposals/job.md

```diff
@@ -0,0 +1,135 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pod(s) that require to run on demand.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+* New Job resource [#7380](https://github.com/GoogleCloudPlatform/kubernetes/pull/7380)
+
+
+## Use Cases
+1. Be able to start of one or several pods tracked as a single entity.
+1. Be able to implement heavy-computation model, eg. MapReduce.
+1. Be able to get the actual job status.
+1. Be able to limit the execution time for a job.
+
+
+## Motivation
+Jobs are needed for executing multi-pod computation "to completion", a good example
+here would be the ability to implement a MapReduce or Hadoop style workload.
+Additionally this new controller should take over pod management logic we currently
+have in certain OpenShift controllers, namely build controller.
```

Comment:
We're not sure if this will work and/or if it's worth the added complexity. For example, while builds and deployments both follow similar conventions for moving their objects from new to pending to running and so on, there is logic that each controller implements alongside the state transitions that is specific to each resource. /cc @smarterclayton @pmorie @derekwaynecarr